### PR TITLE
[#10786] API jars and feature code for MP Rest Client 2.0

### PR DESCRIPTION
**Pull request by andymc12**
Initial check-in for MP Rest Client 2.0 (Epic Issue #10786) - this does not include any implementation yet - just the feature and APIs (based off 2.0-RC2)

### CODE REVIEW

**Comment by andymc12** [comment]
#build

**Comment by andymc12** [comment]
@OpenLiberty/message-reviewer - no need to review, as these messages are all exact copies of the messages shipped with MP Rest Client 1.4.

**Comment by andymc12** [comment]
Previous build failed with what looks like an infrastructure issue... new #build

**Comment by andymc12** [comment]
Tweaked the feature file - new #build

File: dev/com.ibm.websphere.appserver.features/visibility/private/io.openliberty.org.eclipse.microprofile.rest.client-2.0.feature [hunk 1]
```diff
@@ -0,0 +1,11 @@
+-include= ~${workspace}/cnf/resources/bnd/feature.props
+symbolicName=com.ibm.websphere.appserver.org.eclipse.microprofile.rest.client-1.4
+singleton=true
+-features=com.ibm.websphere.appserver.javax.cdi-2.0, \
+ com.ibm.websphere.appserver.javax.annotation-1.3, \
+ com.ibm.websphere.appserver.javax.jaxrs-2.1, \
+ com.ibm.websphere.appserver.org.eclipse.microprofile.config-2.0
```
Comment (tevans78) [inline dev/com.ibm.websphere.appserver.features/visibility/private/io.openliberty.org.eclipse.microprofile.rest.client-2.0.feature:7]: That isn't the right feature name. Should be `io.openliberty.org.eclipse.microprofile.config-2.0`

File: dev/com.ibm.websphere.appserver.features/visibility/private/io.openliberty.org.eclipse.microprofile.rest.client-2.0.feature [hunk 2]
```diff
@@ -0,0 +1,11 @@
+-include= ~${workspace}/cnf/resources/bnd/feature.props
+symbolicName=com.ibm.websphere.appserver.org.eclipse.microprofile.rest.client-1.4
```
Comment (tevans78) [inline dev/com.ibm.websphere.appserver.features/visibility/private/io.openliberty.org.eclipse.microprofile.rest.client-2.0.feature:2]: Should be `io.openliberty...`

File: dev/com.ibm.websphere.appserver.features/visibility/private/io.openliberty.org.eclipse.microprofile.rest.client-2.0.feature [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+singleton=true
+-features=com.ibm.websphere.appserver.javax.cdi-2.0, \
+ com.ibm.websphere.appserver.javax.annotation-1.3, \
+ com.ibm.websphere.appserver.javax.jaxrs-2.1, \
+ com.ibm.websphere.appserver.org.eclipse.microprofile.config-2.0
+-bundles=com.ibm.websphere.org.eclipse.microprofile.rest.client.2.0; location:="dev/api/stable/,lib/"; mavenCoordinates="org.eclipse.microprofile.rest.client:microprofile-rest-client-api:1.4-RC2"
```
Comment (tevans78) [inline dev/com.ibm.websphere.appserver.features/visibility/private/io.openliberty.org.eclipse.microprofile.rest.client-2.0.feature:8]: should be `io.openliberty....`

Comment (tevans78) [reply dev/com.ibm.websphere.appserver.features/visibility/private/io.openliberty.org.eclipse.microprofile.rest.client-2.0.feature:8]: and maven co-ords are slightly wrong?

File: dev/com.ibm.websphere.appserver.features/visibility/public/mpRestClient-2.0/com.ibm.websphere.appserver.mpRestClient-2.0.feature [hunk 4]
```diff
@@ -0,0 +1,25 @@
+-include= ~${workspace}/cnf/resources/bnd/feature.props
+symbolicName=com.ibm.websphere.appserver.mpRestClient-1.4
```
Comment (tevans78) [inline dev/com.ibm.websphere.appserver.features/visibility/public/mpRestClient-2.0/com.ibm.websphere.appserver.mpRestClient-2.0.feature:2]: `io.openliberty...`

**Comment by tevans78** [comment]
I'm going to stop adding comments, I guess it just isn't ready.

**Comment by andrea-pichler** [comment]
No message review requested from L2

**Comment by andymc12** [comment]
#build

**Comment by andymc12** [comment]
1.4 feature verifier was failing - most likely because the 1.4 API bundle was pulling in the 2.0 API jars because it did not specify EXACT.  Updated PR with EXACT text - new #build

**Comment by andymc12** [comment]
Failed to build due to missing com.ibm.ws.require.java8 bundle.... removing from feature and submitting new #build

**Comment by andymc12** [comment]
@tevans78 can you take another look now?  I think all of the issues you found have been addressed.

**Comment by andymc12** [comment]
#run-libby-bot

File: dev/com.ibm.websphere.appserver.features/visibility/public/mpRestClient-2.0/io.openliberty.mpRestClient-2.0.feature [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+IBM-API-Package: \
+  org.eclipse.microprofile.rest.client; type="stable", \
+  org.eclipse.microprofile.rest.client.annotation; type="stable", \
+  org.eclipse.microprofile.rest.client.ext; type="stable", \
+  org.eclipse.microprofile.rest.client.inject; type="stable", \
+  org.eclipse.microprofile.rest.client.spi; type="stable"
+IBM-ShortName: mpRestClient-2.0
+Subsystem-Name: MicroProfile Rest Client 2.0
+-features=io.openliberty.org.eclipse.microprofile.rest.client-2.0, \
+ com.ibm.websphere.appserver.javax.cdi-2.0, \
```
Comment (WhiteCat22) [inline dev/com.ibm.websphere.appserver.features/visibility/public/mpRestClient-2.0/io.openliberty.mpRestClient-2.0.feature:16]: cdi is added by the private feature, does it need to be in both?

**Pull request merged**